src/charm.py

```diff
@@ -775,6 +807,43 @@ def _peers(self) -> Relation:
         """
         return self.model.get_relation(PEER)
 
+    def push_tls_files_to_workload(self) -> None:
+        """Uploads TLS files to the workload container."""
+        key, ca, cert = self.tls.get_tls_files()
+        if key is not None:
+            self._patroni.render_file(f"{self._storage_path}/{TLS_KEY_FILE}", key, 0o600)
+        if ca is not None:
+            self._patroni.render_file(f"{self._storage_path}/{TLS_CA_FILE}", ca, 0o600)
+        if cert is not None:
+            self._patroni.render_file(f"{self._storage_path}/{TLS_CERT_FILE}", cert, 0o600)
+
+        self.update_config()
+
+    def _restart(self, _) -> None:
+        """Restart PostgreSQL."""
+        try:
+            self._patroni.restart_postgresql()
+        except RetryError as e:
+            logger.error("failed to restart PostgreSQL")
```

Comment:
Having a stack trace in the log would be useful, but if there's a security risk there then forget it.

Reply:
Yeah, it's good to have that. Changed on https://github.com/canonical/postgresql-operator/pull/30/commits/c4875f6afef7d294dbd22119e4859f9496b8de86.